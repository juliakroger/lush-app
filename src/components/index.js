export { default as Icon } from './Icon/Icon'
export { default as SubscriptionButton } from './SubscriptionButton/SubscriptionButton'
export { default as PaymentsAccepted } from './PaymentsAccepted/PaymentsAccepted'
