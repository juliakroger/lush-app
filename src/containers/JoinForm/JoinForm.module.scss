@import '../../scss/variables.scss';
@import '../../scss/responsive-utils';

.loginForm {
  display: flex;
  flex-direction: column;
  position: relative;
  min-height: 400px;
  width: 80%;
  max-width: 500px;
  margin: 0 auto;
  text-align: center;
  margin-top: 30px;

  input {
    margin: 0 auto;
    width: 100%;
  }

  .error {
    padding-left: 20px;
    width: 100%;
    color: $super-orange-color;
    font-size: $font-size-small;
  }
}
