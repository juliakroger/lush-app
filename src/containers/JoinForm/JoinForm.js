import React, { useState } from 'react'
import PropTypes from 'prop-types'

import { Input, Button } from '../../components'
import styles from './JoinForm.module.scss'

const JoinForm = ({
  name,
  setName,
  phone,
  setPhone,
  email,
  setEmail,
  emailConfirmation,
  setEmailConfirmation,
  password,
  setPassword,
  passwordConfirmation,
  setPasswordConfirmation,
  submit
}) => {
  const [error, setError] = useState(null)

  const checkRequiredValues = () =>
    !name || !email || !emailConfirmation || !password || !passwordConfirmation

  const hasSomeError = () => {
    if (email && emailConfirmation && email !== emailConfirmation) {
      setError("Emails don't match.")
    }

    if (password && passwordConfirmation && password !== passwordConfirmation) {
      setError("Passwords don't match.")
    }
  }

  return (
    <div className={styles.loginForm}>
      <h1>Create new Account</h1>
      <Input
        placeholder='Your name'
        value={name}
        valueChange={value => setName(value)}
        id='#join-name'
        label='Name *'
        type='text'
      />
      <Input
        placeholder='Phone'
        value={phone}
        valueChange={value => setPhone(value)}
        id='#join-phone'
        label='Phone'
        type='text'
      />
      <Input
        placeholder='amy@example.com'
        value={email}
        valueChange={value => setEmail(value)}
        id='#signin-email'
        label='Email *'
        type='email'
        blurButtonFunction={hasSomeError}
      />
      <Input
        placeholder='amy@example.com'
        value={emailConfirmation}
        valueChange={value => setEmailConfirmation(value)}
        id='#signin-email-confirmation'
        label='Email Confirmation *'
        type='email'
        blurButtonFunction={hasSomeError}
      />

      <Input
        value={password}
        valueChange={value => setPassword(value)}
        id='#signin-password'
        label='Password *'
        type='password'
        blurButtonFunction={hasSomeError}
      />

      <Input
        value={passwordConfirmation}
        valueChange={value => setPasswordConfirmation(value)}
        id='#signin-password'
        label='Password Confirmation *'
        type='password'
        blurButtonFunction={hasSomeError}
      />
      <Button
        text='sign in'
        type={checkRequiredValues() ? 'buttonLight' : 'buttonMint'}
        id='#login-button'
        clickFunction={submit}
      />
      {error ? <span className={styles.error}>{error}</span> : null}
    </div>
  )
}

JoinForm.propTypes = {
  name: PropTypes.any,
  phone: PropTypes.any,
  email: PropTypes.any,
  emailConfirmation: PropTypes.any,
  password: PropTypes.any,
  passwordConfirmation: PropTypes.any,
  setName: PropTypes.func.isRequired,
  setPhone: PropTypes.func.isRequired,
  setEmail: PropTypes.func.isRequired,
  setEmailConfirmation: PropTypes.func.isRequired,
  setPassword: PropTypes.func.isRequired,
  setPasswordConfirmation: PropTypes.func.isRequired,
  submit: PropTypes.func.isRequired
}
export default JoinForm
