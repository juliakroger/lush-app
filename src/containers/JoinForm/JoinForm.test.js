import React from 'react'
import { shallow } from 'enzyme'
import { Button, Input } from '../../components'
import JoinForm from './JoinForm'

describe('Container - Join Form', () => {
  it('should render correctly with all props', () => {
    const component = shallow(
      <JoinForm
        name=''
        setName={jest.fn()}
        phone=''
        setPhone={jest.fn()}
        email=''
        setEmail={jest.fn()}
        emailConfirmation=''
        setEmailConfirmation={jest.fn()}
        password=''
        setPassword={jest.fn()}
        passwordConfirmation=''
        setPasswordConfirmation={jest.fn()}
        submit={jest.fn()}
      />
    )
    expect(component).toMatchSnapshot()
  })

  it('should trigger name input change', () => {
    let name = ''

    const component = shallow(
      <JoinForm
        name=''
        setName={value => (name = value)}
        phone=''
        setPhone={jest.fn()}
        email=''
        setEmail={jest.fn()}
        emailConfirmation=''
        setEmailConfirmation={jest.fn()}
        password=''
        setPassword={jest.fn()}
        passwordConfirmation=''
        setPasswordConfirmation={jest.fn()}
        submit={jest.fn()}
      />
    )
    expect(component).toMatchSnapshot()

    component
      .find(Input)
      .at(0)
      .props()
      .valueChange('nameTest')

    expect(name).toBe('nameTest')
  })

  it('should trigger phone input change', () => {
    let phone = ''

    const component = shallow(
      <JoinForm
        name=''
        setName={jest.fn()}
        phone=''
        setPhone={value => (phone = value)}
        email=''
        setEmail={jest.fn()}
        emailConfirmation=''
        setEmailConfirmation={jest.fn()}
        password=''
        setPassword={jest.fn()}
        passwordConfirmation=''
        setPasswordConfirmation={jest.fn()}
        submit={jest.fn()}
      />
    )
    expect(component).toMatchSnapshot()

    component
      .find(Input)
      .at(1)
      .props()
      .valueChange('phonetest')

    expect(phone).toBe('phonetest')
  })

  it('should trigger email input change', () => {
    let email = ''

    const component = shallow(
      <JoinForm
        name=''
        setName={jest.fn()}
        phone=''
        setPhone={jest.fn()}
        email=''
        setEmail={value => (email = value)}
        emailConfirmation=''
        setEmailConfirmation={jest.fn()}
        password=''
        setPassword={jest.fn()}
        passwordConfirmation=''
        setPasswordConfirmation={jest.fn()}
        submit={jest.fn()}
      />
    )
    expect(component).toMatchSnapshot()

    component
      .find(Input)
      .at(2)
      .props()
      .valueChange('emailtest')

    expect(email).toBe('emailtest')
  })

  it('should trigger emailConfirmation input change', () => {
    let emailConfirmation = ''

    const component = shallow(
      <JoinForm
        name=''
        setName={jest.fn()}
        phone=''
        setPhone={jest.fn()}
        email=''
        setEmail={jest.fn()}
        emailConfirmation=''
        setEmailConfirmation={value => (emailConfirmation = value)}
        password=''
        setPassword={jest.fn()}
        passwordConfirmation=''
        setPasswordConfirmation={jest.fn()}
        submit={jest.fn()}
      />
    )
    expect(component).toMatchSnapshot()

    component
      .find(Input)
      .at(3)
      .props()
      .valueChange('emailtest')

    expect(emailConfirmation).toBe('emailtest')
  })

  it('should trigger password input change', () => {
    let password = ''

    const component = shallow(
      <JoinForm
        name=''
        setName={jest.fn()}
        phone=''
        setPhone={jest.fn()}
        email=''
        setEmail={jest.fn()}
        emailConfirmation=''
        setEmailConfirmation={jest.fn()}
        password=''
        setPassword={value => (password = value)}
        passwordConfirmation=''
        setPasswordConfirmation={jest.fn()}
        submit={jest.fn()}
      />
    )
    expect(component).toMatchSnapshot()

    component
      .find(Input)
      .at(4)
      .props()
      .valueChange('password')

    expect(password).toBe('password')
  })

  it('should trigger passwordConfirmation input change', () => {
    let passwordConfirmation = ''

    const component = shallow(
      <JoinForm
        name=''
        setName={jest.fn()}
        phone=''
        setPhone={jest.fn()}
        email=''
        setEmail={jest.fn()}
        emailConfirmation=''
        setEmailConfirmation={jest.fn()}
        password=''
        setPassword={jest.fn()}
        passwordConfirmation=''
        setPasswordConfirmation={value => (passwordConfirmation = value)}
        submit={jest.fn()}
      />
    )
    expect(component).toMatchSnapshot()

    component
      .find(Input)
      .at(5)
      .props()
      .valueChange('passwordtest')

    expect(passwordConfirmation).toBe('passwordtest')
  })

  it('should trigger submit button click', () => {
    const mockFunc = jest.fn()

    const component = shallow(
      <JoinForm
        name=''
        setName={jest.fn()}
        phone=''
        setPhone={jest.fn()}
        email=''
        setEmail={jest.fn()}
        emailConfirmation=''
        setEmailConfirmation={jest.fn()}
        password=''
        setPassword={jest.fn()}
        passwordConfirmation=''
        setPasswordConfirmation={jest.fn()}
        submit={mockFunc}
      />
    )

    component
      .find(Button)
      .props()
      .clickFunction()

    expect(mockFunc.mock.calls.length).toBe(1)
  })
})
