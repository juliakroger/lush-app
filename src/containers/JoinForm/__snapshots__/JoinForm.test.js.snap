// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container - Join Form should render correctly with all props 1`] = `
<div
  className="loginForm"
>
  <h1>
    Create new Account
  </h1>
  <Input
    blurButtonFunction={[Function]}
    id="#join-name"
    isRequired={false}
    label="Name *"
    placeholder="Your name"
    secondaryLabel={null}
    type="text"
    value=""
    valueChange={[Function]}
  />
  <Input
    blurButtonFunction={[Function]}
    id="#join-phone"
    isRequired={false}
    label="Phone"
    placeholder="Phone"
    secondaryLabel={null}
    type="text"
    value=""
    valueChange={[Function]}
  />
  <Input
    blurButtonFunction={[Function]}
    id="#signin-email"
    isRequired={false}
    label="Email *"
    placeholder="amy@example.com"
    secondaryLabel={null}
    type="email"
    value=""
    valueChange={[Function]}
  />
  <Input
    blurButtonFunction={[Function]}
    id="#signin-email-confirmation"
    isRequired={false}
    label="Email Confirmation *"
    placeholder="amy@example.com"
    secondaryLabel={null}
    type="email"
    value=""
    valueChange={[Function]}
  />
  <Input
    blurButtonFunction={[Function]}
    id="#signin-password"
    isRequired={false}
    label="Password *"
    placeholder=""
    secondaryLabel={null}
    type="password"
    value=""
    valueChange={[Function]}
  />
  <Input
    blurButtonFunction={[Function]}
    id="#signin-password"
    isRequired={false}
    label="Password Confirmation *"
    placeholder=""
    secondaryLabel={null}
    type="password"
    value=""
    valueChange={[Function]}
  />
  <Button
    clickFunction={[MockFunction]}
    id="#login-button"
    text="sign in"
    type="buttonLight"
  />
</div>
`;

exports[`Container - Join Form should trigger email input change 1`] = `
<div
  className="loginForm"
>
  <h1>
    Create new Account
  </h1>
  <Input
    blurButtonFunction={[Function]}
    id="#join-name"
    isRequired={false}
    label="Name *"
    placeholder="Your name"
    secondaryLabel={null}
    type="text"
    value=""
    valueChange={[Function]}
  />
  <Input
    blurButtonFunction={[Function]}
    id="#join-phone"
    isRequired={false}
    label="Phone"
    placeholder="Phone"
    secondaryLabel={null}
    type="text"
    value=""
    valueChange={[Function]}
  />
  <Input
    blurButtonFunction={[Function]}
    id="#signin-email"
    isRequired={false}
    label="Email *"
    placeholder="amy@example.com"
    secondaryLabel={null}
    type="email"
    value=""
    valueChange={[Function]}
  />
  <Input
    blurButtonFunction={[Function]}
    id="#signin-email-confirmation"
    isRequired={false}
    label="Email Confirmation *"
    placeholder="amy@example.com"
    secondaryLabel={null}
    type="email"
    value=""
    valueChange={[Function]}
  />
  <Input
    blurButtonFunction={[Function]}
    id="#signin-password"
    isRequired={false}
    label="Password *"
    placeholder=""
    secondaryLabel={null}
    type="password"
    value=""
    valueChange={[Function]}
  />
  <Input
    blurButtonFunction={[Function]}
    id="#signin-password"
    isRequired={false}
    label="Password Confirmation *"
    placeholder=""
    secondaryLabel={null}
    type="password"
    value=""
    valueChange={[Function]}
  />
  <Button
    clickFunction={[MockFunction]}
    id="#login-button"
    text="sign in"
    type="buttonLight"
  />
</div>
`;

exports[`Container - Join Form should trigger emailConfirmation input change 1`] = `
<div
  className="loginForm"
>
  <h1>
    Create new Account
  </h1>
  <Input
    blurButtonFunction={[Function]}
    id="#join-name"
    isRequired={false}
    label="Name *"
    placeholder="Your name"
    secondaryLabel={null}
    type="text"
    value=""
    valueChange={[Function]}
  />
  <Input
    blurButtonFunction={[Function]}
    id="#join-phone"
    isRequired={false}
    label="Phone"
    placeholder="Phone"
    secondaryLabel={null}
    type="text"
    value=""
    valueChange={[Function]}
  />
  <Input
    blurButtonFunction={[Function]}
    id="#signin-email"
    isRequired={false}
    label="Email *"
    placeholder="amy@example.com"
    secondaryLabel={null}
    type="email"
    value=""
    valueChange={[Function]}
  />
  <Input
    blurButtonFunction={[Function]}
    id="#signin-email-confirmation"
    isRequired={false}
    label="Email Confirmation *"
    placeholder="amy@example.com"
    secondaryLabel={null}
    type="email"
    value=""
    valueChange={[Function]}
  />
  <Input
    blurButtonFunction={[Function]}
    id="#signin-password"
    isRequired={false}
    label="Password *"
    placeholder=""
    secondaryLabel={null}
    type="password"
    value=""
    valueChange={[Function]}
  />
  <Input
    blurButtonFunction={[Function]}
    id="#signin-password"
    isRequired={false}
    label="Password Confirmation *"
    placeholder=""
    secondaryLabel={null}
    type="password"
    value=""
    valueChange={[Function]}
  />
  <Button
    clickFunction={[MockFunction]}
    id="#login-button"
    text="sign in"
    type="buttonLight"
  />
</div>
`;

exports[`Container - Join Form should trigger name input change 1`] = `
<div
  className="loginForm"
>
  <h1>
    Create new Account
  </h1>
  <Input
    blurButtonFunction={[Function]}
    id="#join-name"
    isRequired={false}
    label="Name *"
    placeholder="Your name"
    secondaryLabel={null}
    type="text"
    value=""
    valueChange={[Function]}
  />
  <Input
    blurButtonFunction={[Function]}
    id="#join-phone"
    isRequired={false}
    label="Phone"
    placeholder="Phone"
    secondaryLabel={null}
    type="text"
    value=""
    valueChange={[Function]}
  />
  <Input
    blurButtonFunction={[Function]}
    id="#signin-email"
    isRequired={false}
    label="Email *"
    placeholder="amy@example.com"
    secondaryLabel={null}
    type="email"
    value=""
    valueChange={[Function]}
  />
  <Input
    blurButtonFunction={[Function]}
    id="#signin-email-confirmation"
    isRequired={false}
    label="Email Confirmation *"
    placeholder="amy@example.com"
    secondaryLabel={null}
    type="email"
    value=""
    valueChange={[Function]}
  />
  <Input
    blurButtonFunction={[Function]}
    id="#signin-password"
    isRequired={false}
    label="Password *"
    placeholder=""
    secondaryLabel={null}
    type="password"
    value=""
    valueChange={[Function]}
  />
  <Input
    blurButtonFunction={[Function]}
    id="#signin-password"
    isRequired={false}
    label="Password Confirmation *"
    placeholder=""
    secondaryLabel={null}
    type="password"
    value=""
    valueChange={[Function]}
  />
  <Button
    clickFunction={[MockFunction]}
    id="#login-button"
    text="sign in"
    type="buttonLight"
  />
</div>
`;

exports[`Container - Join Form should trigger password input change 1`] = `
<div
  className="loginForm"
>
  <h1>
    Create new Account
  </h1>
  <Input
    blurButtonFunction={[Function]}
    id="#join-name"
    isRequired={false}
    label="Name *"
    placeholder="Your name"
    secondaryLabel={null}
    type="text"
    value=""
    valueChange={[Function]}
  />
  <Input
    blurButtonFunction={[Function]}
    id="#join-phone"
    isRequired={false}
    label="Phone"
    placeholder="Phone"
    secondaryLabel={null}
    type="text"
    value=""
    valueChange={[Function]}
  />
  <Input
    blurButtonFunction={[Function]}
    id="#signin-email"
    isRequired={false}
    label="Email *"
    placeholder="amy@example.com"
    secondaryLabel={null}
    type="email"
    value=""
    valueChange={[Function]}
  />
  <Input
    blurButtonFunction={[Function]}
    id="#signin-email-confirmation"
    isRequired={false}
    label="Email Confirmation *"
    placeholder="amy@example.com"
    secondaryLabel={null}
    type="email"
    value=""
    valueChange={[Function]}
  />
  <Input
    blurButtonFunction={[Function]}
    id="#signin-password"
    isRequired={false}
    label="Password *"
    placeholder=""
    secondaryLabel={null}
    type="password"
    value=""
    valueChange={[Function]}
  />
  <Input
    blurButtonFunction={[Function]}
    id="#signin-password"
    isRequired={false}
    label="Password Confirmation *"
    placeholder=""
    secondaryLabel={null}
    type="password"
    value=""
    valueChange={[Function]}
  />
  <Button
    clickFunction={[MockFunction]}
    id="#login-button"
    text="sign in"
    type="buttonLight"
  />
</div>
`;

exports[`Container - Join Form should trigger passwordConfirmation input change 1`] = `
<div
  className="loginForm"
>
  <h1>
    Create new Account
  </h1>
  <Input
    blurButtonFunction={[Function]}
    id="#join-name"
    isRequired={false}
    label="Name *"
    placeholder="Your name"
    secondaryLabel={null}
    type="text"
    value=""
    valueChange={[Function]}
  />
  <Input
    blurButtonFunction={[Function]}
    id="#join-phone"
    isRequired={false}
    label="Phone"
    placeholder="Phone"
    secondaryLabel={null}
    type="text"
    value=""
    valueChange={[Function]}
  />
  <Input
    blurButtonFunction={[Function]}
    id="#signin-email"
    isRequired={false}
    label="Email *"
    placeholder="amy@example.com"
    secondaryLabel={null}
    type="email"
    value=""
    valueChange={[Function]}
  />
  <Input
    blurButtonFunction={[Function]}
    id="#signin-email-confirmation"
    isRequired={false}
    label="Email Confirmation *"
    placeholder="amy@example.com"
    secondaryLabel={null}
    type="email"
    value=""
    valueChange={[Function]}
  />
  <Input
    blurButtonFunction={[Function]}
    id="#signin-password"
    isRequired={false}
    label="Password *"
    placeholder=""
    secondaryLabel={null}
    type="password"
    value=""
    valueChange={[Function]}
  />
  <Input
    blurButtonFunction={[Function]}
    id="#signin-password"
    isRequired={false}
    label="Password Confirmation *"
    placeholder=""
    secondaryLabel={null}
    type="password"
    value=""
    valueChange={[Function]}
  />
  <Button
    clickFunction={[MockFunction]}
    id="#login-button"
    text="sign in"
    type="buttonLight"
  />
</div>
`;

exports[`Container - Join Form should trigger phone input change 1`] = `
<div
  className="loginForm"
>
  <h1>
    Create new Account
  </h1>
  <Input
    blurButtonFunction={[Function]}
    id="#join-name"
    isRequired={false}
    label="Name *"
    placeholder="Your name"
    secondaryLabel={null}
    type="text"
    value=""
    valueChange={[Function]}
  />
  <Input
    blurButtonFunction={[Function]}
    id="#join-phone"
    isRequired={false}
    label="Phone"
    placeholder="Phone"
    secondaryLabel={null}
    type="text"
    value=""
    valueChange={[Function]}
  />
  <Input
    blurButtonFunction={[Function]}
    id="#signin-email"
    isRequired={false}
    label="Email *"
    placeholder="amy@example.com"
    secondaryLabel={null}
    type="email"
    value=""
    valueChange={[Function]}
  />
  <Input
    blurButtonFunction={[Function]}
    id="#signin-email-confirmation"
    isRequired={false}
    label="Email Confirmation *"
    placeholder="amy@example.com"
    secondaryLabel={null}
    type="email"
    value=""
    valueChange={[Function]}
  />
  <Input
    blurButtonFunction={[Function]}
    id="#signin-password"
    isRequired={false}
    label="Password *"
    placeholder=""
    secondaryLabel={null}
    type="password"
    value=""
    valueChange={[Function]}
  />
  <Input
    blurButtonFunction={[Function]}
    id="#signin-password"
    isRequired={false}
    label="Password Confirmation *"
    placeholder=""
    secondaryLabel={null}
    type="password"
    value=""
    valueChange={[Function]}
  />
  <Button
    clickFunction={[MockFunction]}
    id="#login-button"
    text="sign in"
    type="buttonLight"
  />
</div>
`;
